### PR TITLE
Msgchanges

### PR DESCRIPTION
**Pull request by Zervox**
changes to Msg::Message and Msg::File prevent duplicate entries when looking for description of items

### CODE REVIEW

File: src/Format/Msg/Message.h [hunk 1]
```diff
@@ -44,24 +44,12 @@ class Message
...
-    std::string text();
-    std::string* textPointer();
-
-    operator std::string();
-
-protected:
-    unsigned int _number = 0;
     std::string _sound;
     std::string _text;
+	std::string _desc;
```
Comment (phobos2077) [inline src/Format/Msg/Message.h]: Are these public properties? Issues:
- Why do you remove getters? 
- Public properties should not start with undescore.
- `_desc` property makes no sense. It duplicates `_text`.

File: src/Format/Msg/Message.h [hunk 2]
```diff
@@ -44,24 +44,12 @@ class Message
 
 public:
     Message();
+	Message(std::string &s, std::string &t);
```
Comment (phobos2077) [inline src/Format/Msg/Message.h]: Why not `const std::string&`? Such interface make the intent of those arguments obscure. In all such cases either use `const std::string&` (read-only use or copy) or `std::string&&` (move). Here const reference is obvious choice.

File: src/Game/ObjectFactory.cpp [hunk 3]
```diff
@@ -135,8 +135,9 @@ Object* ObjectFactory::createObject(unsigned int PID)
             auto msg = ResourceManager::getInstance()->msgFileType("text/english/game/pro_item.msg");
             try
             {
-                object->setName(msg->message(proto->messageId())->text());
-                object->setDescription(msg->message(proto->messageId() + 1)->text());
+				auto o = msg->message(proto->messageId());
```
Comment (phobos2077) [inline src/Game/ObjectFactory.cpp]: Don't use tabs.

File: src/Format/Msg/File.h [hunk 4]
```diff
@@ -50,11 +50,11 @@ class File : public Dat::Item
     File(std::ifstream* stream);
     ~File();
 
-    std::vector<Message*>* messages();
     Message* message(unsigned int number);
 
+
+    std::map<uint32_t,Message*> _messages;
```
Comment (phobos2077) [inline src/Format/Msg/File.h]: Why not `unordered_map`? Also this property should not be public.

**Comment by phobos2077** [comment]
There aren't any actual duplicates in vanilla `msg` files, but the check for this is nice to have for robustness.

**Comment by Zervox** [comment]
actually there is, when it _initializes it checks for files and their descriptor data, the description data gets put into MSG+1 value, meaning you have to look up two memory locations for the same data(and the only area which is checked at runtime is when getting description and name of things you are looking for.

File: src/Game/ObjectFactory.cpp (hunk 3 above)
Comment (Zervox) [inline src/Game/ObjectFactory.cpp]: I am not using tabs, I am using enter, VS2017 does it whenever I reinsert removed lines.

**Comment by phobos2077** [comment]
I think you don't understand how MSG files work. Each message ID is separate, there is no inherent semantic connection between them. The fact that names of objects and their description go together - is just how developers organized their messages. But they should be read individually and each message has only 3 data fields - number, sound and text. Nothing more.

**Pull request closed without merging**